src/ui/bookmarks.ts

```diff
@@ -0,0 +1,227 @@
+import * as vscode from "vscode";
+import * as logging from "@cmt/logging";
+import { TargetNode, BaseNode, DirectoryNode, SourceFileNode } from "@cmt/ui/projectOutline/projectOutline";
+
+const log = logging.createLogger("bookmarks");
+
+export interface BookmarkedItem {
+    id: string;
+    name: string;
+    projectName: string;
+    folderName: string; // Store folder name instead of path for display
+    type: string;
+    sourceNode?: BaseNode; // Store original node reference (not persisted)
+}
+
+export class BookmarkNode extends BaseNode {
+    constructor(public readonly bookmark: BookmarkedItem, id: string) {
+        super(id);
+    }
+
+    getChildren(): BaseNode[] {
+        // Return the children from the original node if it exists
+        if (this.bookmark.sourceNode) {
+            return this.bookmark.sourceNode.getChildren();
+        }
+        return [];
+    }
+
+    getTreeItem(): vscode.TreeItem {
+        const item = new vscode.TreeItem(
+            `${this.bookmark.name} (${this.bookmark.projectName || this.bookmark.folderName})`
+        );
+
+        // Set collapsible state based on whether there are children
+        if (this.getChildren().length > 0) {
+            item.collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
+        } else {
+            item.collapsibleState = vscode.TreeItemCollapsibleState.None;
+        }
+
+        item.tooltip = `${this.bookmark.name}\nProject: ${this.bookmark.projectName || "N/A"}\nType: ${this.bookmark.type}\nFolder: ${this.bookmark.folderName}`;
```

Comment:
This is on the edge of needing to be localized, the only string that would get localized would be `Project`, so while it's only one string, we should likely still localize